src/ai/susi/server/api/susi/SusiService.java

```diff
@@ -182,9 +183,37 @@ public ServiceResponse serviceImpl(Query post, HttpServletResponse response, Aut
                         languageName = groupName.getJSONObject(language);
                         if (languageName.has(skill_name)) {
                             skillName = languageName.getJSONObject(skill_name);
-                            if (skillName.getJSONObject("configure").getBoolean("enable_default_skills") == false) {
+                            Boolean allowed_site = true;
+                            JSONObject configureName = skillName.getJSONObject("configure");
+                            if (configureName.getBoolean("enable_default_skills") == false) {
                                 include_default_skills = false;
                             }
+                            if (configureName.getBoolean("allow_bot_only_on_own_sites") && configureName.has("allowed_sites") && configureName.getString("allowed_sites").length() > 0) {
```

Comment:
This is exactly the same code as in the other class above. We should not do this kind of copy-paste coding because if we must fix a bug somewhere and that is inside such a place, we almost certainly forget to fix the bug in all the affected places.

Please make the functionality of this code a bit abstract, put it into a method and place the method where it belongs most. Then call that method from both servlets.

That is actually a basic thing. Why does this copy-paste coding appear again and again?